app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/individual/BusinessStartDateController.scala

```diff
@@ -93,17 +95,23 @@ class BusinessStartDateController @Inject()(mcc: MessagesControllerComponents,
   private def next(id: String, isEditMode: Boolean, isGlobalEdit: Boolean) = Redirect(
     if (isEditMode || isGlobalEdit) {
       routes.SelfEmployedCYAController.show(id, isEditMode = isEditMode, isGlobalEdit = isGlobalEdit)
+    } else if (isEnabled(StartDateBeforeLimit)) {
+      routes.AddressLookupRoutingController.checkAddressLookupJourney(id, isEditMode)
     } else {
       routes.BusinessTradeNameController.show(id)
     }
   )
 
   //save & retrieve on should have an order of: business name -> business start date (this) -> business trade
   //save & retrieve off should have an order of: business start date (this) -> business name -> business trade
-  def backUrl(id: String, isEditMode: Boolean, isGlobalEdit: Boolean): String = if (isEditMode || isGlobalEdit) {
-    routes.SelfEmployedCYAController.show(id, isEditMode = isEditMode, isGlobalEdit = isGlobalEdit).url
-  } else {
-    routes.BusinessNameController.show(id, isEditMode = isEditMode).url
+  def backUrl(id: String, isEditMode: Boolean, isGlobalEdit: Boolean): String = {
+    if (isEditMode || isGlobalEdit) {
+      routes.SelfEmployedCYAController.show(id, isEditMode = isEditMode, isGlobalEdit = isGlobalEdit).url
+    } else if (isEnabled(StartDateBeforeLimit)) {
+      routes.FullIncomeSourceController.show(id, isEditMode, isGlobalEdit).url
+    } else {
+      routes.BusinessNameController.show(id, isEditMode = isEditMode).url
+    }
```

Comment:
Thinking about this in the bigger picture, i'm not sure if this is 100% correct for the backUrl.
In the scenario that the feature switch "StartDateBeforeLimit" is enabled, if they are in edit mode, they will have come from the CYA page, clicked change on start date which would take them to the streamline page and then they click "No" to enter their own start date.

So I think when the feature switch is enabled, the only place it should go back to is the streamline page.